src/blk/spdk/NVMEDevice.cc

```diff
@@ -569,15 +575,18 @@ int NVMEManager::try_get(const spdk_nvme_transport_id& trid, SharedDriverData **
 
   if (!dpdk_thread.joinable()) {
     dpdk_thread = std::thread(
-      [this, coremask_arg, m_core_arg, mem_size_arg]() {
+      [this, coremask_arg, m_core_arg, mem_size_arg, pci_addr]() {
         static struct spdk_env_opts opts;
+        static struct spdk_pci_addr addr = pci_addr;
```

Comment:
why do we need to use a static variable for `spdk_env_opts` and `spdk_pci_addr`? i think `spdk_env_init()` is a synchronized function, in the sense that we are not obliged to keep its parameter alive after it returns.

Reply:
pci_addr is const, so we use another variable, we can change the type, and remove the static.